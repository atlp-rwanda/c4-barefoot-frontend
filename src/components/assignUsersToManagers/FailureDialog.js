import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import{ useDispatch, useSelector }from 'react-redux';
import { assignUsersFromQueue, cancelAllQueue } from '../../redux/actions/managerSelectedActions';

export default function FailureDialog(props) {
  const retryAssign = useSelector(state => state.addAssignActionToQueue);
  const dispatch = useDispatch(); // for resetting the real [user to manager] state
  const handleTryAgain = () => {
    assignUsersFromQueue(dispatch, retryAssign.pendingTasks);
  }
  const handleCancel = () => {
    cancelAllQueue(dispatch);
  }
  return (
    <Dialog
      open={props.open}
      onClose={()=>{}}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title" style={{color: '#EB5757'}}>{"Error!"}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description" style={{color: '#EB5757'}}>
          Could not assign user(s) to manager(s)
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleTryAgain} color="primary">
          TRY AGAIN
        </Button>
        <Button onClick={handleCancel} style={{color: '#333333'}}>
          CANCEL
        </Button>
      </DialogActions>
    </Dialog>
  );
}
