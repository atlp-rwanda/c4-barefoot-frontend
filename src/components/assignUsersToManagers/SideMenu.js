import React from 'react';
import SideDrawer from '../SideDrawer';

const managerLinks = [
    {title: 'Home', path: '/' },
    {title: 'My Travel', path: '/' },
    {title: 'Report requests', path: '/' },
    {title: 'Book accommodation', path: '/' },
    {title: 'Users', path: '/' }
]


const SideMenu = () => {
    return (
        <SideDrawer navLinks={managerLinks} />
    )
};

export default SideMenu;
