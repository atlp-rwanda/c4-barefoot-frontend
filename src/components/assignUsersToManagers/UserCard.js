import { Avatar, Card, CardContent, Typography, Button, CardActions,
    Select, MenuItem } from '@material-ui/core';
import { Skeleton } from '@material-ui/lab';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addUsersToAssignQueue } from '../../redux/actions/managerSelectedActions';

export const UserCardSkeleton = () => {
  return (
    <Card style={{margin: 15}}>
      <Skeleton width={200} height={200} variant='circle' />
      <Typography variant='h5'><Skeleton /></Typography>
      <Typography variant='h5'><Skeleton /></Typography>
      <Typography variant='h5' style={{width: '50%', float: 'right'}}><Skeleton /></Typography>
    </Card>
  )
};

const UserCard = (props) => {
    const { skeleton, managersElements, user: USER } = props;
    const cardStyle = {maxWidth: 250, margin: 2, padding: 3, backgroundColor: '#EAF4FB'};
    const dispatch = useDispatch();
    // const assignUsersToManagersQueue = useSelector(state => state.addAssignActionToQueue);
    // console.log(assignUsersToManagersQueue);
    const handleManagerSelected = ({target}) => {
      const managerId = target.value;
      dispatch(addUsersToAssignQueue(USER.id, managerId));
    }
    return (
      skeleton
    ? <UserCardSkeleton />
    : <Card style={cardStyle}>
        <Avatar 
            style={{width: '100%', maxHeight: '50%', minHeight: 250, backgroundColor: '#ABD5ED'}}
            alt="user" 
            name='profile_picture' 
            margin='normal'
            src={USER.profile_picture ? USER.profile_picture : 'https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg' }
        />
        <Typography gutterBottom component="h2" style={{fontWeight: 600, color: '#43A0D6'}}>
            {`${USER.first_name} ${USER.last_name}`}
        </Typography>
        <Select onChange={handleManagerSelected} style={{width: '100%'}} defaultValue={props.manager_id ? props.manager_id : 'none'}>
          <MenuItem value="none" disabled>
            <Avatar />
            <Typography><em>assign line manager</em></Typography>
          </MenuItem>
           {managersElements} 
        </Select>
        <CardActions>
          <Button href={`/${props.username}`} size="small" color="primary" style={{marginLeft: 'auto', marginTop: 10}}>
            Profile
          </Button>
        </CardActions>
    </Card>
    )
};

export  default UserCard;
