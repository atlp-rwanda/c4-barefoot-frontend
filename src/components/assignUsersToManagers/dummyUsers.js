export const USERS = {
    "count": 8,
    "rows": [
        {
            "id": "a9610cf3-4056-41dd-92ca-463088e23d07",
            "first_name": "Manager",
            "last_name": "Manager",
            "email": "With_LineManager@gmail.com",
            "username": "With_LineManager",
            "occupation": "manager_occupation",
            "bio": null,
            "verified": true,
            "user_role_id": "7254a9e7-2e1b-4f83-ad73-78b90dd3df77",
            "manager_id": "fb94de4d-47ff-4079-89e8-b0186c0a3be8",
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-25T06:49:46.281Z"
        },
        {
            "id": "fb94de4d-47ff-4079-89e8-b0186c0a3be8",
            "first_name": "Manager",
            "last_name": "Two",
            "email": "managertwo@gmail.com",
            "username": "managertwo",
            "occupation": "manager_occupation",
            "bio": null,
            "verified": true,
            "user_role_id": "7254a9e7-2e1b-4f83-ad73-78b90dd3df77",
            "manager_id": "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-25T06:49:46.281Z"
        },
        {
            "id": "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
            "first_name": "Manager",
            "last_name": "One",
            "email": "mj@gmail.com",
            "username": "managerOne",
            "occupation": "manager_occupation",
            "bio": null,
            "verified": true,
            "user_role_id": "7254a9e7-2e1b-4f83-ad73-78b90dd3df77",
            "manager_id": "fb94de4d-47ff-4079-89e8-b0186c0a3be8",
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-25T06:49:46.281Z"
        },
        {
            "id": "d74fcc5e-5755-4366-83ef-cf306b013c46",
            "first_name": "pushRequester",
            "last_name": "One",
            "email": "pushnotfication@gmail.com",
            "username": "pushrequester",
            "occupation": "push_notification_requester",
            "bio": null,
            "verified": true,
            "user_role_id": "45429837-ed2c-435d-bc22-ad9c5dbe3782",
            "manager_id": "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-25T06:49:46.281Z"
        },
        {
            "id": "38eb202c-3f67-4eed-b7ac-9c31bc226e0c",
            "first_name": "Line",
            "last_name": "Manager",
            "email": "manager_id@gmail.com",
            "username": "manager_id",
            "occupation": "manager_occupation",
            "bio": null,
            "verified": true,
            "user_role_id": "7254a9e7-2e1b-4f83-ad73-78b90dd3df77",
            "manager_id": null,
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-28T17:05:40.149Z"
        },
        {
            "id": "83b2a3e7-9ba4-4d3f-b3a3-d31940ee2edc",
            "first_name": "Requester",
            "last_name": "One",
            "email": "sequester@gmail.com",
            "username": "requesterOne",
            "occupation": "requester_occupation",
            "bio": null,
            "verified": true,
            "user_role_id": "45429837-ed2c-435d-bc22-ad9c5dbe3782",
            "manager_id": "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-25T06:50:00.573Z"
        },
        {
            "id": "122a0d86-8b78-4bb8-b28f-8e5f7811c456",
            "first_name": "SUper",
            "last_name": "Admin",
            "email": "superadmin@gmail.com",
            "username": "superadmin",
            "occupation": "admin_occupation",
            "bio": null,
            "verified": true,
            "user_role_id": "eface43c-2e49-473b-bbe2-305d1a5190f1",
            "manager_id": null,
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-25T08:36:24.468Z"
        },
        {
            "id": "2d647115-3af7-4df0-99aa-6656c764829f",
            "first_name": "Travel",
            "last_name": "Admin",
            "email": "traveladmin@gmail.com",
            "username": "travelAdmin",
            "occupation": "travelAdmin_occupation",
            "bio": null,
            "verified": true,
            "user_role_id": "4fd084a0-cdd6-47a5-aaf5-5fdc8b5629dd",
            "manager_id": null,
            "profile_picture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
            "language": "English",
            "address": "Kigali",
            "createdAt": "2021-02-25T06:49:46.281Z",
            "updatedAt": "2021-02-25T08:36:59.113Z"
        }
    ]
};

export const MANAGERS = {

};
