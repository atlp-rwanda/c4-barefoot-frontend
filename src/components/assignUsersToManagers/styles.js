import { makeStyles } from '@material-ui/core/styles';

const styles = makeStyles((theme) => ({
    center: {
        textAlign: 'center'
    }, 


    Login: {
      height: '90vh',
    },
    paper: {
      margin: 'auto',
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justify: 'center',
      textAlign: 'center',
    },
    paper2: {
      margin: 'auto',
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justify: 'center',
      textAlign: 'center',
      color: 'white',
      height: '90%'
    },
    form: {
      width: '100%', 
      marginTop: theme.spacing(1),
    },
    login: {
      margin: theme.spacing(3, 0, 2),
      width: '40%'
    },
    submit: {
      margin: theme.spacing(3, 0, 12),
      width: '35%'
    },
    social_media: {
      margin: theme.spacing(3, 0, 2),
      width: '40%',
      height:'40px'
    },
    image_icon:{
        width: theme.spacing(3),
        height: theme.spacing(3)
    },
    social_media_sm: {
      margin: theme.spacing(3, 0, 2),
      width: '40%',
      height:'40px'
    },
    social_media_grid: {
      margin: 'auto',
      marginTop: '0',
      width: '80%',
      justifyContent: 'center',
      alignItems: 'center',
      textAlign: 'center'
    },
    secondpart: {
      margin: theme.spacing(5, 0, 5),
      width: '60%',
      justifyContent: 'center'
    },
    forgotPassword: {
      display: 'flex',
      justifyContent: 'flex-end',
    },
    
  }));

export default styles;
