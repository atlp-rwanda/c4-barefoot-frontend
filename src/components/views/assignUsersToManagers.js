import React from 'react';
import AssignUsers from '../assignUsersToManagers';

export default function AssignUsersToManagers(){
  return <AssignUsers />;
}