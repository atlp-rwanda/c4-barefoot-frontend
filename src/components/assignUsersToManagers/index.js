import React, { useEffect } from 'react';
import { Container } from '@material-ui/core';
import { connect, useDispatch } from 'react-redux';
import UsersList from './UsersList';
import { getManagersList, getVerifiedUsers } from '../../redux/actions/assignUserActions';
import FailureDialog from './FailureDialog';
import SuccessDialog from './SuccessDialog';

const AssignUsersToManagers = (props) => {
  const { addAssignActionToQueue, fetchVerifiedUsers, fetchAllManagers } = props;
  const successDialog = true, failureDialog = !successDialog;
  const pending = (fetchAllManagers.pending && fetchVerifiedUsers.pending)
                  && !fetchAllManagers.loaded;
  console.log({fetchVerifiedUsers,fetchAllManagers, addAssignActionToQueue});
  const dispatch = useDispatch();
  useEffect(() => {
    props.getVerifiedUsers(dispatch);
    props.getManagersList(dispatch);
  }, []);

  return (
    <Container style={{display: 'flex', flexDirection: 'row', justify: 'flex-start', padding: 'unset'}}>
      <UsersList loading={pending} />
        <SuccessDialog open={successDialog} />
        <FailureDialog open={failureDialog}/>
    </Container>
  );
}

const mapStateToProps = state => ({
  fetchVerifiedUsers: state.fetchVerifiedUsers,
  fetchAllManagers: state.fetchAllManagers,
  addAssignActionToQueue: state.addAssignActionToQueue
});

export { AssignUsersToManagers };

export default connect(mapStateToProps, { getVerifiedUsers, getManagersList })(AssignUsersToManagers);
