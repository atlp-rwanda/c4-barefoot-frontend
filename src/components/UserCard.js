import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Card, FormControl,TextField,  CardActionArea, CardActions, CardContent} from '@material-ui/core';
import {CardMedia, Button, Typography, IconButton, Avatar, Modal} from '@material-ui/core';
import { Delete, Email, Place, Language, Work, SupervisedUserCircle, AccountCircle } from '@material-ui/icons'
import { connect } from 'react-redux'
import { getUsers } from '../redux/actions/fetchUsersAction';
import { Skeleton } from '@material-ui/lab';


const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 265,
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    backgroundColor: '#EAF4FB',
  },
  media: {
    height: 140
  },
  actions: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  cardContent: {
    display: 'flex',
    flexDirection: 'column',
    color: '#43A0D6'
  },
  modal: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
  paper: {
    backgroundColor: "#EAF4FB",
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),
    outline: "none",
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  modalAvatar: {
    width: theme.spacing(12),
    height: theme.spacing(12)
  },
  modalItems: {
    display: "flex",
    justifyContent: 'center'
    
    
  }
  
}));


function UserCard(props) {
  
  const classes = useStyles();

  const [open, setOpen] = React.useState(false);

  const openProfile = () => {
    setOpen(true)
  }

  const closeProfile = () => {
    setOpen(false)
  }

  const ProfileModal = (
      <Modal className={classes.modal} open={open} onClose={closeProfile}>
        <div className={classes.paper}>
          <Avatar
            alt="profile picture"
            src={props.UserData && (props.UserData.profile_picture)}
            className={classes.modalAvatar}
          />
          <div className={classes.modalItems}>
            <AccountCircle />
            <Typography> {props.UserData && (props.UserData.first_name)} {props.UserData && (props.UserData.last_name)}</Typography>
          </div>
          <div className={classes.modalItems}>
            <Email/>
            <Typography> {props.UserData && (props.UserData.email)} </Typography>
          </div>
          <div className={classes.modalItems}>
            <Place />
            <Typography>{props.UserData && (props.UserData.address)}</Typography>
          </div>
          <div className={classes.modalItems}>
            <Language />
            <Typography>{props.UserData && (props.UserData.language)}</Typography>
          </div>
          <div className={classes.modalItems}>
            <Work />
            <Typography>Software Enginner</Typography>
          </div>
          <div className={classes.modalItems}>
            <SupervisedUserCircle />
            <Typography> Line manager</Typography>
          </div>
        </div>
      </Modal>
  )

  
  return (
    <Card className={classes.root}>
      <CardActionArea>
        {props.pending ? 
        (<Skeleton variant='rect' className={classes.media} />)
        :
        (<CardMedia
          className={classes.media}
          image='https://i.ibb.co/MGMCsqm/profile-Picture.png'
        />)}
        <CardContent className={classes.cardContent}>
          <Typography gutterBottom variant="subtitle1">
            {props.pending ? (<Skeleton width='100%'/>)
            :
            (props.UserData.first_name)+ " "+ (props.UserData.last_name)}
          </Typography>
          <FormControl>
            <TextField
            select
            id='role'
            label='assign role'
            />
          </FormControl>
          <FormControl>
            <TextField
            select
            label = 'assign line manager'
            />
          </FormControl>
        </CardContent>
      </CardActionArea>
      <CardActions className={classes.actions}>
        <IconButton>
          <Delete color='secondary'/>
        </IconButton>
        <Button size="small" color="primary" onClick={openProfile}> Profile </Button>
      </CardActions>
      {ProfileModal}
    </Card>
  );
}

const mapStateToProps = state => ({
  pending : state.users.pending
})

export default connect(mapStateToProps, {getUsers})(UserCard)
