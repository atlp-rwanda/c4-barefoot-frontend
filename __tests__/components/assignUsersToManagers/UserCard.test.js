import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { UserCardSkeleton } from '../../../src/components/assignUsersToManagers/UserCard';

describe('<UserCardSkeleton />', ()=> {
    let wrapper;
    it( 'It should render without crashing', ()=>{
        wrapper= shallow(<UserCardSkeleton />)
        const component= toJson(wrapper);
        expect(component).toMatchSnapshot();
    });
});
