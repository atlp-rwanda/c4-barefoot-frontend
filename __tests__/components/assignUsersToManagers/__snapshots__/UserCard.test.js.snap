// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCardSkeleton /> It should render without crashing 1`] = `
<WithStyles(ForwardRef(Card))
  style={
    Object {
      "margin": 15,
    }
  }
>
  <WithStyles(ForwardRef(Skeleton))
    height={200}
    variant="circle"
    width={200}
  />
  <WithStyles(ForwardRef(Typography))
    variant="h5"
  >
    <WithStyles(ForwardRef(Skeleton)) />
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    variant="h5"
  >
    <WithStyles(ForwardRef(Skeleton)) />
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    style={
      Object {
        "float": "right",
        "width": "50%",
      }
    }
    variant="h5"
  >
    <WithStyles(ForwardRef(Skeleton)) />
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Card))>
`;
